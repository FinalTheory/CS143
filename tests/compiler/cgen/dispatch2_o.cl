-- dispatch on another class

class Alpha
{
   niam() : Int {90};
};

class Main inherits IO
{
   func() : Int {(new Alpha).niam()};
   main() : Object {{ out_int(func()); self; }};
};
