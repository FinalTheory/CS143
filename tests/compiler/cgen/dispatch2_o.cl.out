SPIM Version 6.5 of January 4, 2003
Copyright 1990-2003 by James R. Larus (larus@cs.wisc.edu).
All Rights Reserved.
See the file README for a full copyright notice.
Loaded: /usr/class/cs143/cool/lib/trap.handler
90COOL program successfully executed
Stats -- #instructions : 261
         #reads : 53  #writes 40  #branches 51  #other 117
